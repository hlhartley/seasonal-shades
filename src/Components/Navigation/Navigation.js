import React from 'react';

export const Navigation = () => {
    return (
        <div className='navigation-container'>
            <button>EYES</button>
            <button>LIPS</button>
            <button>CHEEKS</button>
            <button>NAILS</button>
        </div>
    )
};