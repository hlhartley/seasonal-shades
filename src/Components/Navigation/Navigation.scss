.navigation-container {
    border-bottom: 15px solid #d77871;
    display: flex;
    justify-content: space-around;
    padding-bottom: 15px;
    
    button{
        border: none;
        font-family: $nav-font;
        font-size: 17px;
        letter-spacing: 1.75px;
    };
};