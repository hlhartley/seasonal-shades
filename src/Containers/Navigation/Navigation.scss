.navigation-container {
    border-bottom: 15px solid $tan-color;
    display: flex;
    justify-content: space-around;
    
    button{
        border: none;
        font-size: 17px;
        letter-spacing: 2.5px;
    };

    a {
        display: flex;
        align-items: center;
        justify-content: center;
        border-top-left-radius: 4px;
        border-top-right-radius: 4px;
        height: 30px;
        margin-bottom: 0;
        background-color: $light-grey-color;
        width: 20vw;
    };

    .is-active {
        background-color: $tan-color;
    };
};

