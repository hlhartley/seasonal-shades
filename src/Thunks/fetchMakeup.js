import { getLipstick, getLipstickColors, setLoading, setError } from '../Actions/index';
import { API } from '../Helpers/requests';

export const fetchMakeup = (path) => {
    return async (dispatch) => {
        try {
            dispatch(setLoading(true));
            const colors = {};
            const items = {};
            const result = await API(path);
            result.forEach((item) => {
                item.product_colors.forEach(color => {
                    // colors[color.hex_value] = item.id;
                    if (!colors[color.hex_value]) {
                        colors[color.hex_value] = [];
                    } 
                    colors[color.hex_value].push(item.id);
                })
                items[item.id] = item;
            });
            switch(path) {
                case 'lipstick':
                    dispatch(getLipstick(items));
                    dispatch(getLipstickColors(colors));
            }
        } catch (error) {
            dispatch(setError(error));
        }
        dispatch(setLoading(false));
    }
}
