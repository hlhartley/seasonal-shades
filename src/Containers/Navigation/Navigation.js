import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchMakeup } from '../../Thunks/fetchMakeup';

export class Navigation extends Component {

    handleClick = async (type) => {
        this.props.updateCurrentType(type)
        await this.props.fetchMakeup(type)
    }

    render() {
        return (
            <div className='navigation-container'>
                <button onClick={() => this.handleClick('eyeshadow')}>EYES</button>
                <button onClick={() => this.handleClick('lipstick')}>LIPS</button>
                <button onClick={() => this.handleClick('blush')}>CHEEKS</button>
                <button onClick={() => this.handleClick('nail_polish')}>NAILS</button>
            </div>
        )
    }
}
    // console.log(props)

const mapDispatchToProps = (dispatch) => ({
    fetchMakeup: (path) => dispatch(fetchMakeup(path)),
});

export default connect(null, mapDispatchToProps)(Navigation)
