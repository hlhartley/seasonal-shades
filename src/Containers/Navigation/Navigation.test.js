import React from 'React';
import { Navigation } from './Navigation';
import { shallow } from 'enzyme';
import { fetchMakeup } from '../../Thunks/fetchMakeup';
import { mapDispatchToProps } from './Navigation';

jest.mock('../../Thunks/fetchMakeup')

describe('Navigation', () => {
    let wrapper;
    describe('Navigation component', () => {
        beforeEach(() => {
            let mockUpdateCurrentType = jest.fn()
            let mockFetchMakeup= jest.fn()
            wrapper = shallow(<Navigation updateCurrentType={mockUpdateCurrentType} lipstickColors={[]} blushColors={undefined} eyeshadowColors={undefined} nail_polishColors={['Blue']} fetchMakeup={mockFetchMakeup}/>)
        });

        it('should match the snapshot with all the data passed in correctly', () => {
            expect(wrapper).toMatchSnapshot()
        });

        it('should have a proper default state', () => {
            expect(wrapper.state()).toEqual({
                activeLink: 'eyeshadow',
            });
        });

        it('handleClick should setState with the activeLink, call updateCurrentType, and call fetchMakeup', async () => {
            wrapper.instance().handleClick('lipstick')
            expect(wrapper.instance().state.activeLink).toEqual('lipstick')
            wrapper.instance().handleClick('nailpolish')
            expect(wrapper.instance().state.activeLink).toEqual('nailpolish')
            expect(wrapper.instance().props.updateCurrentType).toHaveBeenCalled()
            await expect(wrapper.instance().props.fetchMakeup).toHaveBeenCalled()
        })
    });

    describe('mapDispatchToProps', async () => {
        it('calls dispatch with fetchMakeup action when fetchMakeup is called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = fetchMakeup('lipstick')
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.fetchMakeup('lipstick')
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        });
    });
})