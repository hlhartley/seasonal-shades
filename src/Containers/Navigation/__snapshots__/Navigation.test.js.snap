// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Navigation component should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="navigation-container"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="is-active"
    onClick={[Function]}
    to="/eyeshadow"
  >
    EYES
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="not-active"
    onClick={[Function]}
    to="/lipstick"
  >
    LIPS
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="not-active"
    onClick={[Function]}
    to="/blush"
  >
    CHEEKS
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="not-active"
    onClick={[Function]}
    to="/nail_polish"
  >
    NAILS
  </NavLink>
</div>
`;
